lightningd/peer_htlcs.c

```diff
@@ -1682,9 +1682,8 @@ void peer_got_revoke(struct channel *channel, const u8 *msg)
 	for (i = 0; i < tal_count(changed); i++) {
 		/* If we're doing final accept, we need to forward */
 		if (changed[i].newstate == RCVD_ADD_ACK_REVOCATION) {
-			if (!peer_accepted_htlc(channel, changed[i].id, false,
-						&failcodes[i]))
-				return;
+			peer_accepted_htlc(channel, changed[i].id, false,
+					   &failcodes[i]);
```

Comment:
I don't quite follow the logic why this check was removed. @rustyrussell could you help me?

Reply:
Just because an htlc failed does *not* mean we should stop processing the rest!

If it's due to an internal error, sure, [eer_accepted_htlc calls channel_internal_error() and we clean up the subdaemon.  But there are some paths which return false and don't call this: in particular, the case where we can't process the onionpacket.

In these cases, we need to respond to channeld, and save things to the db.  Otherwise it just sits there waiting for a reply we never send.